addons/udes_stock/tests/test_refactoring_splitting.py

```diff
@@ -762,3 +762,118 @@ def test01_auto_unlink_empty_pickings(self):
+        self.picking_type_pick.write(
+            {
+                "u_post_assign_action": "by_maximum_quantity",
+                "u_assign_refactor_constraint_value": 2,
+            }
+        )
+
+        self.picking = self.create_picking(self.picking_type_pick)
+
+    def test_split_equally(self):
+        """Check that a pick is split according to maximum quantity"""
+        Picking = self.env["stock.picking"]
+
+        # Create apple quant
+        self.create_quant(self.apple.id, self.test_location_01.id, 10)
+        self.create_move(self.apple, 10, self.picking)
+        self.picking.action_assign()
+
+        # Get pickings
+        all_pickings = Picking.search([("picking_type_id", "=", self.picking_type_pick.id)])
+        # Check each picking is only for 2 products
+        self.assertEqual(len(all_pickings), 5)
+        self.assertTrue(
+            all([sum(pick.move_lines.mapped("product_uom_qty")) == 2 for pick in all_pickings])
+        )
+
+    def test_split_unequally(self):
+        """Check that a pick is split according to maximum quantity
+        when it doesn't divide equally
+        """
+        Picking = self.env["stock.picking"]
+
+        # Create apple quant
+        self.create_quant(self.apple.id, self.test_location_01.id, 9)
+        self.create_move(self.apple, 9, self.picking)
+        self.picking.action_assign()
+
+        # Get pickings
+        all_pickings = Picking.search([("picking_type_id", "=", self.picking_type_pick.id)])
+        # Check pickings are split correctly
+        self.assertEqual(len(all_pickings), 5)
+        quantities_per_pick = [
+            sum(pick.move_lines.mapped("product_uom_qty")) for pick in all_pickings
+        ]
+        self.assertCountEqual(quantities_per_pick, [2, 2, 2, 2, 1])
+
+    def test_split_partial_reserve(self):
+        """Check that a pick is split according to maximum quantity
+        with any unreserved quantities moved to a seprate picking
+        """
+        Picking = self.env["stock.picking"]
+
+        # Create apple quant
+        self.create_quant(self.apple.id, self.test_location_01.id, 7)
+        self.create_move(self.apple, 10, self.picking)
+        self.picking.action_assign()
+
+        # Get pickings
+        all_pickings = Picking.search([("picking_type_id", "=", self.picking_type_pick.id)])
+        # Check pickings are split correctly
+        self.assertEqual(len(all_pickings), 5)
+        quantities_ordered_and_reserved = [
+            (
+                sum(pick.move_lines.mapped("product_uom_qty")),
+                sum(pick.move_lines.mapped("reserved_availability"))
+            ) for pick in all_pickings
+        ]
+        expected_quantities_ordered_and_reserved = [
+            (2, 2), (2, 2), (2, 2), (1, 1), (3, 0)
+        ]
+        self.assertCountEqual(
+            quantities_ordered_and_reserved, expected_quantities_ordered_and_reserved
+        )
+
+    def test_split_mixed_products(self):
```

Comment:
Can we get a similar unit test but the moves are for more just to cover that extra case